packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -116,229 +159,791 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeamBuildSetting({
-    final String? savedCertChoice = config.getValue('ios-signing-cert') as String?;
+  /// Select code-signinging settings and save to config.
+  ///
+  /// Available options include automatic signing with a code-signing identity
+  /// or manual code-signing with a provisioning profile.
+  Future<void> selectSettings() async {
+    // If terminal UI can't be used, just attempt with the first valid certificate
+    // since we can't ask the user.
+    if (!_terminal.stdinHasTerminal) {
+      _logger.printError(
+        'Unable to detect stdin for the terminal. Code-signing selection requires stdin.',
+      );
+      return;
+    }
+    _terminal.usesTerminalUi = true;
+
+    final bool toolsValidated = await _validateCodeSignSearchTools(printError: true);
+    if (!toolsValidated) {
+      return;
+    }
 
-    if (savedCertChoice != null) {
-      if (validCodeSigningIdentities.contains(savedCertChoice)) {
-        logger.printStatus(
-          'Found saved certificate choice "$savedCertChoice". To clear, use "flutter config".',
-        );
-        return savedCertChoice;
-      } else {
-        logger.printError(
-          'Saved signing certificate "$savedCertChoice" is not a valid development certificate',
+    final String? savedCertChoice =
+        _config.getValue(XcodeCodeSigningSettings.kConfigCodeSignCertificate) as String?;
+    final String? savedProfile =
+        _config.getValue(XcodeCodeSigningSettings.kConfigCodeSignProvisioningProfile) as String?;
+
+    if (savedCertChoice != null || savedProfile != null) {
+      _logger.printError(
+        'Code-signing settings are already set. To reset them, use "flutter config '
+        '--clear-ios-signing-settings"',
+      );
+      return;
+    }
+
+    final _CodeSigningStyle? style = await _selectSigningStyle();
+    if (style == null) {
+      _logger.printWarning(_codeSignSelectionCanceled);
+      return;
+    }
+
+    if (style == _CodeSigningStyle.automatic) {
+      final List<String> validCodeSigningIdentities = await _getSigningIdentities();
+      if (validCodeSigningIdentities.isEmpty) {
+        _logger.printError(noCertificatesInstruction, emphasis: true);
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
+      }
+      final String? identity = await _selectSigningIdentity(validCodeSigningIdentities);
+      if (identity == null) {
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
+      }
+    } else if (style == _CodeSigningStyle.manual) {
+      final List<_ProvisioningProfile> validProvisioningProfiles = await _getProvisioningProfiles();
+      if (validProvisioningProfiles.isEmpty) {
+        _logger.printError(
+          'No provisioning profiles were found. To learn how to create or download '
+          'a provisioning profile, please see '
+          'https://developer.apple.com/help/account/manage-provisioning-profiles/create-a-development-provisioning-profile',
```

Comment:
Do you think it'd be worth creating a `flutter.dev/to/create-provisioning-profile` redirect link so that if Apple changes their docs we can update this link for folks that are still on old Flutter tool versions? See: https://github.com/flutter/flutter/blob/master/docs/contributing/use-reliable-links.md

If you go forward with this, don't forget to update line 853 below as well